### PR TITLE
update dependencies and tests

### PR DESCRIPTION
**Pull request by mimmi20**
This PR wants:
- update dependencies to actual versions (for the supported PHP versions)
- remove some dependencies which were not used, or were replaced by using docker
- reformat `dependabot.yml`, add a section to update Github actions
- remove deprecated composer update option `--no-suggest`
- update coding style definition, disable the `after_heredoc` options as this will crash for PHP < 7.3
- fix coding style issues and issues found by phpstan
- update tests

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,42 +1,62 @@
...
-  - dependency-name: symfony/finder
-    versions:
-    - ">= 0"
-  - dependency-name: twig/twig
-    versions:
-    - ">= 0"
-  commit-message:
-    prefix: Build
-    include: scope
+  - package-ecosystem: composer
```
Comment (asgrim) [inline .github/dependabot.yml:6]: yikes, were the line endings `\r\n` before? O_O

File: .php_cs [hunk 2]
```diff
@@ -23,8 +23,13 @@ return PhpCsFixer\Config::create()
         '@PHP71Migration' => true,
         '@PHP71Migration:risky' => true,
         '@PHP73Migration' => true,
+        '@PHP74Migration' => true,
```
Comment (asgrim) [inline .php_cs:26]: All this CS maintenance things we should switch to `doctrine/coding-standard`, so much simpler than maintaining this ruleset I think .... thoughts?

**Review by asgrim — APPROVED** [review]
Hmm is there anything we can do to stop codecov "failing" the build? I think it's just because coverage dropped or something, IDK if that should be a build-fail reason?

File: .php_cs (hunk 2 above)
Comment (mimmi20) [reply .php_cs:26]: All 3 projects (`browscap`, `browscap-php` and `browscap-site`) have their own coding styles and tooling sets. If we change the coding standard, we should unify the coding standard and the tools over all projects (in a seperate PR).

Comment (asgrim) [reply .php_cs:26]: I am ok with that; `doctrine/coding-standard` is easy to pull in & apply (`vendor/bin/phpcs` to check, `vendor/bin/phpcbf` to update).

**Comment by mimmi20** [comment]
> Hmm is there anything we can do to stop codecov "failing" the build? I think it's just because coverage dropped or something, IDK if that should be a build-fail reason?

I dont think that this is an issue, as long we dont define these tests as `required`.

**Comment by asgrim** [comment]
> I dont think that this is an issue, as long we dont define these tests as `required`.

Fair enough :) just on glance, all I see is "failed build" so I tend to ignore it heh

**Pull request merged**